src/VisualStudio/Core/Def/Implementation/ChangeSignature/AddParameterDialogViewModel.cs

```diff
@@ -226,7 +226,8 @@ private void SetCurrentTypeTextAndUpdateBindingStatus(string typeName)
         private bool IsParameterTypeSyntacticallyValid(string typeName)
         {
             var languageService = Document.GetRequiredLanguageService<IChangeSignatureViewModelFactoryService>();
-            return languageService.IsTypeNameValid(typeName);
+            _ = Document.TryGetSyntaxTree(out var syntaxTree);
+            return languageService.IsTypeNameValid(typeName, syntaxTree?.Options);
```

Comment:
yeah, this is really hacky.  why do we need options to know if a type name is valid?

Reply:
Because if the langversion isn't set appropriately, the parser will error because the feature isn't available.

Reply:
comment left above on how i think we should do this isntead.  would def want thsi line undone.

Reply:
Now passing latest.